Sources/Action.swift

```diff
@@ -155,12 +169,32 @@ public final class Action<Input, Output, Error: Swift.Error> {
 			}
 
 			disposable += {
-				self._isExecuting.value = false
+				self.state.modify {
+					$0.isExecuting = false
+				}
 			}
 		}
 	}
 }
 
+private struct ActionState {
+	var isExecuting: Bool
+	var value: Any
+	private let userEnabled: (Any) -> Bool
+
+	init(isExecuting: Bool, value: Any, isEnabled: @escaping (Any) -> Bool) {
+		self.isExecuting = isExecuting
+		self.value = value
+		self.userEnabled = isEnabled
+	}
+
+	/// Whether the action should be enabled for the given combination of user
+	/// enabledness and executing status.
+	fileprivate var isEnabled: Bool {
+		return userEnabled(value) && !isExecuting
```

Comment:
We should cache the value of `userEnabled(value)` and only call it when the value changes.